meta-mender-tegra/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-qspi-sd/flash_mender.xml

```diff
@@ -48,21 +48,22 @@
               binary. </description>
         </partition>
 
-        <!-- This is padding to ensure VER is at the end of flash -->
-        <partition name="PAD" id="6" type="data">
+        <partition name="UBENV" type="data">
             <allocation_policy> sequential </allocation_policy>
             <filesystem_type> basic </filesystem_type>
-            <size> 3342336 </size>
+	    <start_location> 0x3B0000 </start_location>
+            <size> 262144 </size>
```

Comment:
> I'm wondering if it would be better to keep UBENV out of VER_b and VER partitions

I played around with different locations for these partitions.  The VER/VER_b partitions are only used by the `l4t_payload_updater_t210` script to compare the current version against the version in the BUP payload.  The data stored in them is very small, on the order of 100 bytes or so, and they're looked up via the pseudo-GPT in the SPI flash, so there's no problem with relocating them and reducing the size (at least for now). Relocating the U-Boot environment block, on the other hand, would affect upgrades/downgrades between pre-32.4.x and post-32.4.x unless we add some state scripts to save and restore at least the `mender_` variables so they're propagated correctly.

So in the end it looked like relocating and shrinking the VER/VER_b partitions was the safer choice.

Reply:
> Relocating the U-Boot environment block, on the other hand, would affect upgrades/downgrades between pre-32.4.x and post-32.4.x

Agreed, however I'm not sure if this really matters as much for the sdcard nano versions since I don't expect anyone is using in production.  So if there's a a tradeoff between future support headaches and breaking upgrades between the dunfell release we just did weeks ago and this version I'd tend to lean toward the breaking upgrades for the sake of future compatibility on this particular platform.  I haven't tried upgrades from zeus to dunfell on this platform myself.

I'm wondering if it would be a good exercise to go back through each platform and look for differences between the stock partition layout and mender layout to make sure we didn't miss synchronizing with any L4T flash layout updates.  It looks like we missed one on this platform at least.  I haven't done this myself, not sure if you or @kekiefer have already.  If not I can look at this.

Reply:
> So if there's a a tradeoff between future support headaches and breaking upgrades between the dunfell release we just did weeks ago and this version I'd tend to lean toward the breaking upgrades for the sake of future compatibility on this particular platform.

I'm fine either way; given what VER is used for, I think this change is fairly low risk, but you never know.

Another option would be for us to reduce the size of the U-Boot environment (by setting `BOOTENV_SIZE` for the SPI/SD platforms) to 0x1800 (96KiB) or even 0x1000 (64KiB) so that two copies can fit between the UBOOTENV start location and VER_b's start location. The default size that Mender assigns is quite large, and we only use about 6KiB in the default environment setup, so it's unlikely that the reduction would have any impact for most use cases.

Reply:
> Another option would be for us to reduce the size of the U-Boot environment (by setting BOOTENV_SIZE for the SPI/SD platforms) to 0x1800 (96KiB) or even 0x1000 (64KiB) so that two copies can fit between the UBOOTENV start location and VER_b's start location

I like this idea the best of all proposed thus far.

Reply:
> I like this idea the best of all proposed thus far.

OK, I'll update the PR to do it that way, then.

> I'm wondering if it would be a good exercise to go back through each platform and look for differences between the stock partition layout and mender layout

Good idea.  I haven't done a comprehensive comparison.

Reply:
> OK, I'll update the PR to do it that way, then.

I've refreshed and rebased the patches for this, using 64KiB for the environment size.